modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/extension/v1_0/test/ObjectEntryExtensionProviderTest.java

```diff
@@ -0,0 +1,231 @@
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.object.rest.internal.extension.v1_0.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.headless.admin.user.dto.v1_0.UserAccount;
+import com.liferay.object.model.ObjectDefinition;
+import com.liferay.object.service.ObjectDefinitionLocalService;
+import com.liferay.object.service.ObjectFieldLocalService;
+import com.liferay.object.util.LocalizedMapUtil;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.test.util.UserTestUtil;
+import com.liferay.portal.kernel.util.DateUtil;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.kernel.util.UnicodePropertiesBuilder;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.util.PropsUtil;
+import com.liferay.portal.vulcan.extension.ExtensionProvider;
+import com.liferay.portal.vulcan.extension.PropertyDefinition;
+
+import java.io.Serializable;
+
+import java.math.BigDecimal;
+
+import java.sql.Timestamp;
+
+import java.util.Collections;
+import java.util.Date;
+import java.util.Map;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Carolina Barbosa
+ */
+@RunWith(Arquillian.class)
+public class ObjectEntryExtensionProviderTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayIntegrationTestRule liferayIntegrationTestRule =
+		new LiferayIntegrationTestRule();
+
+	@BeforeClass
+	public static void setUpClass() throws Exception {
+		PropsUtil.addProperties(
+			UnicodePropertiesBuilder.setProperty(
+				"feature.flag.LPS-135404", "true"
+			).build());
+
+		ObjectDefinition objectDefinition =
+			_objectDefinitionLocalService.fetchObjectDefinitionByClassName(
+				TestPropsValues.getCompanyId(), User.class.getName());
+
+		_addCustomObjectField(
+			objectDefinition.getObjectDefinitionId(), "Boolean", "Boolean",
+			"booleanField", false);
+		_addCustomObjectField(
+			objectDefinition.getObjectDefinitionId(), "Date", "Date",
+			"dateField", true);
+		_addCustomObjectField(
+			objectDefinition.getObjectDefinitionId(), "Decimal", "Double",
+			"decimalField", false);
+		_addCustomObjectField(
+			objectDefinition.getObjectDefinitionId(), "PrecisionDecimal",
+			"BigDecimal", "precisionDecimalField", true);
+
+		_user = UserTestUtil.addUser();
```

Comment:
Don't we have to get rid of these fields and user?

Reply:
I tested and these fields and this user are automatically deleted after running this test

Reply:
Great. Thanks. 🙏🏻